File: client/src/app/service/adventure.service.ts
import { UserService } from './user.service';
import { HttpClient } from '@angular/common/http';
import { ConfigService } from './config.service';
import { ApiService } from './api.service';
import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import {FreeEntityDTO} from "../model/free-entity-dto";
import {CottageDTO} from "../model/cottage-dto.model";
import {AdventureDTO} from "../model/adventure-dto";

@Injectable({
  providedIn: 'root'
})
export class AdventureService {

  adventure: any;

  constructor(
    private apiService: ApiService,
    private config: ConfigService,
    private httpClient: HttpClient
  ) { }

  getAdventure(id:number){
    return this.apiService.get(this.config.adventure_url + '/get_adventure/' + id);
  }

  updateAdventure(id:number,data:any){
    return this.apiService.post(this.config.adventure_url + '/update_adventure/' + id, data);
  }

  saveAdventure(data:any){
    return this.apiService.post(this.config.adventure_url + '/add_adventure', data);
  }

  saveImage(data:any, id:number){
    return this.httpClient.post(this.config.adventure_url + '/save_adventure_image/'+id, data);
  }

  deleteAdventure(id:number){
    return this.apiService.delete(this.config.adventure_url + '/delete_adventure/'+id);
  }
  getFreeAdventures(request: FreeEntityDTO) {
    return this.apiService.post(this.config.adventure_url + '/findFreeAdventures',  JSON.parse(JSON.stringify(request)));
  }
  getSorted(adventures: AdventureDTO[], criterion: string, asc: boolean){
    return this.apiService.post('http://localhost:8090/adventures/sorted' + '/' + criterion + '/' + asc, JSON.parse(JSON.stringify(adventures)));
  }
}

File: isa2/demo/src/main/java/isa2/demo/Service/AdventureService.java
package isa2.demo.Service;

import isa2.demo.DTO.AdventureDTO;
import isa2.demo.DTO.FreeEntityDTO;
import isa2.demo.Exception.InvalidInputException;
import isa2.demo.Model.Adventure;
import isa2.demo.Model.Owner;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.Optional;

public interface AdventureService {

    List<Adventure> findAll();

    Optional<Adventure> findOne(Integer id);

    Adventure save(Adventure adventure);

    Adventure update(Adventure adventure);

    void delete(Integer id);

    List<Adventure> findAdventuresByInstructor(Owner owner);

    Collection<Adventure> findFreeAdventures(FreeEntityDTO request) throws InvalidInputException;

    ArrayList<AdventureDTO> sortAdventures(Collection<AdventureDTO> adventures, String criterion, boolean asc);

    Owner getOwnerForAdventure(Integer adventureId);
}
